vendor.conf

```diff
@@ -6,6 +6,7 @@ github.com/cpuguy83/go-md2man a65d4d2de4d5f7c74868dfa9b202a3c8be315aaa
 github.com/davecgh/go-spew 346938d642f2ec3594ed81d874461961cd0faa76
 github.com/docker/distribution edc3ab29cdff8694dd6feb85cfeb4b5f1b38ed9c
 github.com/docker/docker a1be987ea9e03e5ebdb1b415a7acdd8d6f0aaa08
+github.com/resin-os/balena b3dd1a619b85181156c2e75e8368a61e0d886fb3
```

Comment:
Note the two lines above: `docker/docker` followed by `resin-os/balena`. Even before the renaming pull requests, the unit tests were failing with this output:
```
cd ~/resin/resin-os/balena-cli  # where I cloned balena-cli
make -f docker.Makefile shell  # open a dev shell in a suitable container
make test-unit  # within the build container, run the unit tests
...
# github.com/docker/cli/cli/command/image
cli/command/image/build.go:383:10: unknown field 'Volumes' in struct literal of type types.ImageBuildOptions
cli/command/image/delta.go:37:27: clnt.ImageDelta undefined (type "github.com/docker/cli/vendor/github.com/docker/docker/client".APIClient has no field or method ImageDelta)
# github.com/docker/cli/cli/command/image
cli/command/image/build.go:383:10: unknown field 'Volumes' in struct literal of type types.ImageBuildOptions
cli/command/image/delta.go:37:27: clnt.ImageDelta undefined (type "github.com/docker/cli/vendor/github.com/docker/docker/client".APIClient has no field or method ImageDelta)
make: *** [Makefile:15: test-unit] Error 2
```
So, for example, in balena-cli's `delta.go`, there is a call to `ImageDelta` which is implemented in `resin-os/balena`, but which is not implemented by the "original" `docker/docker` (which Docker renamed to `moby/moby`, it seems).
I got the unit tests to pass by replacing the _contents_ of `docker/docker` with the contents of `resin-os/balena`, but I wonder how best to proceed. For example:
* We could commit the replaced contents of `docker/docker`, leaving a mismatched commit hash in `vendor.conf`, knowing/accepting for now that the `vndr` tool would not work correctly for `docker/docker`. On this point, I have noticed that `vndr` downloads ~15MB for `resin-os/balena` versus ~2MB of the current `docker/docker`.
* We could delete `docker/docker` from `vendor.conf`, leave `resin-os/balena` instead, then change hundreds of import `docker/docker` occurrences in the source code of balena-cli, to import `resin-os/balena` instead. This may be the cleanest solution, if it doesn't lead to unexpected (to me) side effects or a snowball of additional changes.
* We could postpone dealing with this issue, and for now leave the unit and e2e tests in a non-passing state. I could add a note to the TESTING.md file that manual replacing of `docker/docker` is required for the tests to pass.
* Something else?

Reply:
@petrosagg ^^^